homeassistant/components/wled/config_flow.py

```diff
@@ -63,6 +66,9 @@ async def _handle_config_flow(
                 return self._show_confirm_dialog()
             return self._show_setup_form()
 
+        # if prepare is True, user_input can not be None.
+        assert user_input is not None
```

Comment:
Asserts should not be used. There is no guarantee the assert is None (or the typing on this method is incorrect).

Reply:
`user_input` cannot be None _here_ based on how this method is currently called. See the check a few lines earlier:

```
        if user_input is None and not prepare:
```

... and if `prepare` is  True, then  user_input won't be None (see `async_step_zeroconf`).  So I think both the method typing is correct (it can be both None and not None), but also this assert not fail.

However, I agree it's not readable. Do you have an alternative in mind? Some options:
   * An `if` statement that will not fire (and isn't easily testable).
   * A cast: `user_input = cast(dict[str, Any], user_input)`
   * Some sort of larger refactoring.
   * ??

Reply:
We should not assert to please typing, that makes no sense. Typing is not about pleasing mypy. I rather leave it unstrict than adding this kind of logic.

Reply:
Is the `cast` any better in your eyes?

Reply:
No? The thing triggers, and that should be addressed codewise, and this probably needs refactoring for it.

Reply:
IMO, this is a valid use case for `assert` because bound between `prepare` and `user_input` can't be expressed using python typing system.

Another option is adding:
```python
if user_input is None:
    return self.async_abort(reason="invalid_input")
```

Reply:
I'll leave it to @frenck to comment on whether the `if` statement (or `assert`) solution is acceptable in exchange for type checking of this integration.

If not, it would require a larger refactor to fix. My fallback will be to just modify the PR to keep the acceptable typing fixes (to make it easer for a future contributor to work on) but disable the enforcement.

Reply:
Let keeps it this way, I've added a task on my list to refactor it out. This assertion doesn't belong here, nor does that if statement.